### PR TITLE
feat: account creation and presence channel

### PR DESCRIPTION
**Pull request by Carl-Tabuso**
### 🛠 Changes (notable only)

- [omnia-digital/livewire-calendar](https://github.com/omnia-digital/livewire-calendar) was added for calendar manager. Install this on your local copy by running `composer install`. I published the blade files in `views/vendor/livewire-calendar/`. The styling uses tailwind and it looks horrible and shit on the calendar page — I made test files on https://github.com/POWERLANE-TEAM/emplo-hrms/pull/77/commits/8d361aad9083d39e08b687d1051733f6af274fe0
- For seeding of `job_titles` `job_levels` `specific_areas` `job_families` I had their seeders modified to match the actual data from Powerlane, cause what's the purpose of using fake data anymore. This is the  [job-titles.json](https://github.com/user-attachments/files/17673384/job-titles.json). Have this stored in `storage/app/public/utils/`.
- Made some renaming of files `BasicRoleSeeder -> BasicUserSeeder` | `IntermediateRoleSeeder -> IntermediateUserSeeder` | `AdvancedRoleSeeder -> AdvancedUserSeeder`. Another thing, I remove the givePermissionTo() method, since the purpose is to assign a role and the user will automatically inherit the permissions of the role he is assigned of.
- In `RolesAndPermissionsSeeder` I made some methods to properly categorize the creation of roles and permissions as well as their initial association — assign initial permissions to each roles.
- I added new backed enums for civil statuses and sexes. Who knows if that might change in the future so that we're not directly making changes to the migration files.
- I created a presence channel called `online-users` in `channel.php`, enabling the client to render lists of online users. This feature is currently only available to advanced users. To ensure that each online users are captured, ensure that you include the `online-users.js` inside your blade or js files that are only accessible to authenticated users.
- The create account form is now working. Not much validation is enforced since i don't really know what to check for, honestly.
- New foreign relationship in the `employees` migration file, i overlooked the shift and employee association. You should run a fresh migration.
- `public fn fullName() { ... }` accessors for applicant and employee are available to use in your blade files, simply `$user->full_name`
- @biellamariscotes  should modify the styles of calendar package to fit in our case.
- New relationships were added in `Barangays` `Provinces` `Cities` and `Regions` so whichever parent model, you should be able to access the relationships, since they are **code-based**
- Newly created migration and model `Announcement` for announcement publishing.
- New command was added in our concurrent script `php artisan queue:work`

### ✔️ Checklist

- [x] Is `staging` the base branch of this PR?
- [x] Do your changes not reveal sensitive information, such as secrets, API keys, etc?
- [x] Are there no erroneous console logs, debuggers, or leftover code in your changes?

### CODE REVIEW

**Comment by C-Ivan-Bandilla** [comment]
> * For seeding of `job_titles` `job_levels` `specific_areas` `job_families` I had their seeders modified to match the actual data from Powerlane, cause what's the purpose of using fake data anymore. This is the  [job-titles.json](https://github.com/user-attachments/files/17673384/job-titles.json). Have this stored in `public/storage/utils/`

I put it it first in public\storage\utils
image


This it what worked on my end 
storage\app\public\utils\job-titles.json

**Comment by C-Ivan-Bandilla** [comment]
The problem that is I had to uncomment the client side validation 

image

**Comment by Carl-Tabuso** [comment]
>This it what worked on my end
storage\app\public\utils\job-titles.json

have u checked if job_titles table after encountering this exception, were they not populated?

> The problem that is I had to uncomment the client side validation

Are these the only allowed domains in emails?
```json
{
    "valid_email": [
        "gmail.com",
        "yahoo.com",
        "outlook.com"
    ],
    "invalid_email": []
}
```

**Comment by C-Ivan-Bandilla** [comment]
> Are these the only allowed domains in emails?
> 
> ```json
> {
>     "valid_email": [
>         "gmail.com",
>         "yahoo.com",
>         "outlook.com"
>     ],
>     "invalid_email": []
> }
> ```

Yes currently

maybe a free email should work in fake seeder

**Comment by C-Ivan-Bandilla** [comment]
> > This it what worked on my end
> > storage\app\public\utils\job-titles.json
> 
> have u checked if job_titles table after encountering this exception, were they not populated?

It is not pupoluted if the json is located at public\storage\utils but

if I placed it in storage\app\public\utils\job-titles.json it worked on my end

**Comment by Carl-Tabuso** [comment]
my instructions were wrong.

image

**Comment by Carl-Tabuso** [comment]
i just load the json file and access the values in `valid_email` array. This should work now

File: resources/js/admin/dashboard.js [hunk 1]
```diff
@@ -20,5 +21,5 @@ document.addEventListener("DOMContentLoaded", (event) => {
 initSidebar();
 
 
-
+zzz
```
Comment (C-Ivan-Bandilla) [inline resources/js/admin/dashboard.js:24]: I think this can be removed

**Review by C-Ivan-Bandilla — APPROVED** [review]
Things I tested are

1. Database seeding
2. Creation of accounts 
3.  Online Users
4. Using fullname attribute

- Calendar values are static values

Suggestion is that in the **creation of account** it would be great if there is a **same as present address checkbox** when filling the permanent address. I forgot where I saw this form feature

**Comment by Carl-Tabuso** [comment]
> Suggestion is that in the creation of account it would be great if there is a same as present address checkbox when filling the permanent address. I forgot where I saw this form feature

this crossed my mind recently but i sort of just forgot about it.

**Pull request merged**